tests/lib/flat-compat.js

```diff
@@ -0,0 +1,995 @@
+        rules: {},
+        environments: {},
+        processors: {},
+        ...plugin
+    };
+}
+
+/**
+ * Returns the full directory path for a fixture directory.
+ * @param {string} dirName The directory name to resolve.
+ * @returns {string} The full directory path to the fixture.
+ */
+function getFixturePath(dirName) {
+    return path.join(FIXTURES_BASE_PATH, dirName);
+}
+
+//-----------------------------------------------------------------------------
+// Tests
+//-----------------------------------------------------------------------------
+
+describe("FlatCompat", () => {
+
+    describe("config()", () => {
+
+        let compat;
+        const baseDirectory = getFixturePath("config");
+        const pluginFixture1 = normalizePlugin(require(path.join(baseDirectory, "node_modules/eslint-plugin-fixture1")));
+        const pluginFixture2 = normalizePlugin(require(path.join(baseDirectory, "node_modules/eslint-plugin-fixture2")));
+        const pluginFixture3 = normalizePlugin(require(path.join(baseDirectory, "node_modules/eslint-plugin-fixture3")));
+
+        beforeEach(() => {
+            compat = new FlatCompat({
+                baseDirectory
+            });
+        });
+
+        describe("top-level", () => {
+
+            describe("ignorePatterns", () => {
+                it("should translate ignorePatterns string into ignores array", () => {
+                    const result = compat.config({
+                        ignorePatterns: "*.jsx"
+                    });
+
+                    assert.strictEqual(result.length, 1);
+                    assert.typeOf(result[0].ignores[0], "function");
+                    assert.isTrue(result[0].ignores[0]("/usr/eslint/foo.jsx"));
+                    assert.isFalse(result[0].ignores[0]("/usr/eslint/foo.js"));
+                });
+
+                it("should translate ignorePatterns array into ignores array", () => {
+                    const result = compat.config({
+                        ignorePatterns: ["*.jsx"]
+                    });
+
+                    assert.strictEqual(result.length, 1);
+                    assert.typeOf(result[0].ignores[0], "function");
+                    assert.isTrue(result[0].ignores[0]("/usr/eslint/foo.jsx"));
+                    assert.isFalse(result[0].ignores[0]("/usr/eslint/foo.js"));
+                });
+
+                it("should ignore second argument of ignore function from ignorePatterns", () => {
+                    const result = compat.config({
+                        ignorePatterns: ["*.jsx"]
+                    });
+
+                    assert.strictEqual(result.length, 1);
+                    assert.typeOf(result[0].ignores[0], "function");
+                    assert.isTrue(result[0].ignores[0]("/usr/eslint/foo.jsx", {}));
+                    assert.isFalse(result[0].ignores[0]("/usr/eslint/foo.js", ""));
+                });
+
+                it("should combine ignorePatterns from extended configs", () => {
```

Comment:
@mysticatea did I set this test up correctly? This is the one that's currently failing.

Reply:
This test is wrong.

In `.gitignore` spec, if a directory is ignored, we cannot unignore the inside of the directory. In this case, the `ignores-foo` config ignores `foo/`, so this config cannot unignore `foo/bar`. If the `ignores-foo` config ignores `foo/*`, i.g. ignoring the child of `foo/`, then this config can unignore `foo/bar`.

Reply:
This is the reason that the default ignore patterns are `/node_modules/*` instead of `/node_modules`. In the past, we received the request that wants to unignore `/node_modules/foo`. (https://github.com/eslint/eslint/issues/5410)

Reply:
Ah that makes sense. I’ll update the test, thanks!

I’m not sure if flat config can do that, I’d need to investigate. Can you open an issue describing that use case?

Reply:
Thanks, that fixed it!

Reply:
Oops, I spoke too soon (my setup was incorrect). The test is still failing even after changing `ignores-foo` to `foo/*`. Is there something else I'm doing wrong?

Reply:
Fixed! :tada:

Reply:
Thank you. I'm sorry for my late response. Yes, `.gitignore` doesn't affect to outside of the directory.

> I’m not sure if flat config can do that, I’d need to investigate. Can you open an issue describing that use case?

I opened: https://github.com/eslint/eslintrc/issues/25